src/ray/core_worker/core_worker.cc

```diff
@@ -103,6 +103,18 @@ CoreWorker::CoreWorker(const WorkerType worker_type, const Language language,
   gcs_client_ = std::make_shared<gcs::RedisGcsClient>(gcs_options);
   RAY_CHECK_OK(gcs_client_->Connect(io_service_));
 
+  // Subscribe to all unexpected worker failure notifications from the local
+  // and remote raylets. Note that this does not include workers that failed
+  // due to node failure.
```

Comment:
Could you expand a bit more on how we plan to handle worker failures that are due to node failure?

Reply:
It seems like to handle the reference count and fate sharing issues mentioned below, we also need to handle worker death in this case?

Reply:
As Eric mentioned, we can use the node ID in a similar way to check which workers died. I'll update the comment to include that.